src/k8s_lb_manager.py

```diff
@@ -0,0 +1,237 @@
+    ):
+        """Initialize the KubernetesLoadBalancer.
+
+        :param name: Name of the LoadBalancer.
+        :param namespace: Namespace for the LoadBalancer.
+        :param ports: List of ServicePort objects.
+        :param additional_labels: Additional labels to apply.
+        :param additional_selectors: Additional selectors to match pods.
+        :param additional_annotations: Additional annotations to apply.
+        """
+        self.name = name
+        self.namespace = namespace
+        self.ports = ports
+        self.field_manager = field_manager
+        self.additional_labels = additional_labels or {}
+        self.additional_selectors = additional_selectors or {}
+        self.additional_annotations = additional_annotations or {}
+        self.additional_annotations = parse_annotations(additional_annotations)
+        # Initialize Kubernetes client
+        self.client = Client(namespace=self.namespace, field_manager=self.field_manager)
+
+        if self._annotations_valid():
+            self.reconcile()
+        else:
+            self.remove_lb()
+
+    def remove_lb(self):
+        """Removes the LoadBalancer."""
+        try:
+            self.client.delete(Service, name=self.name, namespace=self.namespace)
+            logger.info(f"Deleted LoadBalancer {self.name} in namespace {self.namespace}")
+        except ApiError as e:
+            logger.info(f"Failed to delete LoadBalancer {self.name}: {e}")
+
+    def reconcile(self):
+        """Reconcile the LoadBalancer's state."""
+        # Desired state of the LoadBalancer
+        service = Service(
+            metadata=ObjectMeta(
+                name=self.name,
+                namespace=self.namespace,
+                labels=self.additional_labels,
+                annotations=self.additional_annotations,
+            ),
+            spec=ServiceSpec(
+                ports=self.ports,
+                selector=self.additional_selectors,
+                type="LoadBalancer",
+            ),
+        )
+
+        try:
+            # Check if the service exists
+            existing_service = self.client.get(Service, name=self.name, namespace=self.namespace)
+
+            # Patch if differences exist
+            if not self._is_patched(old_service=existing_service, new_service=service):
+                self.client.patch(Service, name=self.name, obj=service, patch_type=PatchType.APPLY)
+                logger.info(f"Patched LoadBalancer {self.name} in namespace {self.namespace}")
+            else:
+                logger.debug(f"No changes for LoadBalancer {self.name}")
+        except ApiError as e:
+            # Create the service if it doesn't exist
+            if e.status.code == 404:
+                self.client.create(service)
+                logger.info(f"Created LoadBalancer {self.name} in namespace {self.namespace}")
+            else:
+                logger.info(f"Failed to create LoadBalancer {self.name}: {e}")
```

Comment:
What about using the [kubernetes resource manager](https://github.com/canonical/lightkube-extensions/blob/main/src/lightkube_extensions/batch/_kubernetes_resource_manager.py) here?  That way you get reconcile and deletion logic for free